### PR TITLE
add GetContent endpoint for projects

### PR DESCRIPTION
**Pull request by irissmann**

### CODE REVIEW

File: src/main/java/com/google/gerrit/extensions/api/projects/BranchApi.java [hunk 1]
```diff
@@ -25,8 +25,17 @@
   void delete() throws RestApiException;
 
   /**
-   * A default implementation which allows source compatibility
```
Comment (uwolfer) [inline src/main/java/com/google/gerrit/extensions/api/projects/BranchApi.java]: Please revert these whitespace changes.

File: src/main/java/com/google/gerrit/extensions/api/projects/BranchApi.java [hunk 2]
```diff
@@ -25,8 +25,17 @@
...
-   * when adding new methods to the interface.
+   * Look up a file by the given path in the HEAD revision of the branch.
+   * 
+   * @param path Path to the file.
+   * @return Returns a instance of <code>FileApi</code> which represents the file.
```
Comment (uwolfer) [inline src/main/java/com/google/gerrit/extensions/api/projects/BranchApi.java]: Please use here also {@code...}.

File: src/main/java/com/google/gerrit/extensions/api/projects/FileApi.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package com.google.gerrit.extensions.api.projects;
+
+import com.google.gerrit.extensions.restapi.BinaryResult;
+import com.google.gerrit.extensions.restapi.NotImplementedException;
+import com.google.gerrit.extensions.restapi.RestApiException;
+
+public interface FileApi {
+
+  /**
+   * Returns the content of the file from the HEAD revision as a Base64 encoded <code>BinaryResult</code>.
```
Comment (uwolfer) [inline src/main/java/com/google/gerrit/extensions/api/projects/FileApi.java]: Please use here also {@code...}.

File: src/main/java/com/urswolfer/gerrit/client/rest/http/projects/BranchApiRestClient.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import com.google.gerrit.extensions.api.projects.FileApi;
+import com.google.gerrit.extensions.restapi.RestApiException;
+import com.google.gson.JsonElement;
+import com.urswolfer.gerrit.client.rest.http.GerritRestClient;
+
+/**
+ * @author Ingo Rissmann
+ */
+public class BranchApiRestClient extends BranchApi.NotImplemented implements BranchApi {
+  private final GerritRestClient gerritRestClient;
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/gerrit/client/rest/http/projects/BranchApiRestClient.java]: Please use 4 spaces indent for this file.

File: src/main/java/com/urswolfer/gerrit/client/rest/http/projects/BranchApiRestClient.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  private final BranchInfoParser branchInfoParser;
+  private final ProjectApiRestClient projectApiRestClient;
+  private final String name;
+
+  public BranchApiRestClient(GerritRestClient gerritRestClient, BranchInfoParser branchInfoParser,
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/gerrit/client/rest/http/projects/BranchApiRestClient.java]: I use one line per parameter for such longer ctors.

File: src/main/java/com/urswolfer/gerrit/client/rest/http/projects/FileApiRestClient.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import com.google.common.base.Suppliers;
+import com.google.gerrit.extensions.api.projects.FileApi;
+import com.google.gerrit.extensions.restapi.BinaryResult;
+import com.google.gerrit.extensions.restapi.RestApiException;
+import com.google.gerrit.extensions.restapi.Url;
+import com.urswolfer.gerrit.client.rest.http.GerritRestClient;
+import com.urswolfer.gerrit.client.rest.http.util.BinaryResultUtils;
+
+public class FileApiRestClient implements FileApi {
+  private final GerritRestClient gerritRestClient;
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/gerrit/client/rest/http/projects/FileApiRestClient.java]: Please use 4 spaces indent for this file.

File: src/main/java/com/urswolfer/gerrit/client/rest/http/projects/ProjectApiRestClient.java [hunk 7]
```diff
@@ -38,10 +39,8 @@
     private final BranchInfoParser branchInfoParser;
     private final String name;
 
-    public ProjectApiRestClient(GerritRestClient gerritRestClient,
-                                ProjectsParser projectsParser,
-                                BranchInfoParser branchInfoParser,
-                                String name) {
+    public ProjectApiRestClient(GerritRestClient gerritRestClient, ProjectsParser projectsParser,
```
Comment (uwolfer) [inline src/main/java/com/urswolfer/gerrit/client/rest/http/projects/ProjectApiRestClient.java]: Please revert these format changes.

**Comment by uwolfer** [comment]
Thanks for your PR. Could you please squash these commits together into one? If you have never done that, please see: http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit

**Comment by uwolfer** [comment]
I have just created a API proposal change for the official Gerrit API:
https://gerrit-review.googlesource.com/73020

It is slightly simplified for @irissmann proposal. Once this change gets merged into Gerrit, I will update this implementation accordingly.

**Comment by uwolfer** [comment]
Squashed and committed commit of @irissmann into master: https://github.com/uwolfer/gerrit-rest-java-client/commit/342acef25b818c2cea236f8e477b8ed2bcb67fe5

Adopted to final Gerrit API: https://github.com/uwolfer/gerrit-rest-java-client/commit/049894bd1521c87ad319d2f6729e9e8e445f4af1 and added some test: https://github.com/uwolfer/gerrit-rest-java-client/commit/2ed49788f22910b0250864a07381d9f2b1da50c0.

**Comment by irissmann** [comment]
Hi Urs
Today I wanted to implement your comments, but I see now that you have already processed the pull request. Sorry for being late :)
Do you know when the next release will be created?

Best regard,
Ingo.

**Comment by uwolfer** [comment]
@irissmann: Thanks for your work!

I think I will create a new release in the next days / weeks. There are also quite some other fixes / improvements included in that release.

**Pull request closed without merging**